### DISCUSSION TITLE
MultiBloc Part 2

### DISCUSSION THREAD

**Discussion by codetricity** [caitlynbakery/theta_x_multi_bloc#4] [Ideas]
![image](https://user-images.githubusercontent.com/5799860/174871647-a4b99cd6-8ce5-4885-bd25-700ca751a806.png)

![image](https://user-images.githubusercontent.com/5799860/174872020-f6322805-799f-4650-8daa-7d3247d70bc8.png)

![image](https://user-images.githubusercontent.com/5799860/174872286-62be710f-bb3f-4e9f-ac86-a7c29bd5e0c1.png)

```dart

  @override
  Widget build(BuildContext context) {
    return BlocBuilder<CameraUseBloc, CameraUseState>(
      builder: (context, state) {
        context.read<CameraUseBloc>().add(GetCameraModeEvent());
        return Column(
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                state.cameraMode == CameraMode.image
```

**Comment by caitlynbakery** [comment]
Ok, I'll implement this idea in Concept #5. Concept #5 will include getting the camera mode when it first starts building the app.

**Comment by jcasman** [comment]
OK, please include the information on getting camera mode in the README. I think this is a key technique for working with the camera.

**Reply by codetricity** [reply]
I think that for the multibloc, she's working on this feature set:
https://docs.google.com/document/d/18wnGAoSNWLVh98c9u-GN9ngL_O0Q45ZkFSaokPLtXiw/edit?usp=sharing

**Comment by jcasman** [comment]
OK, understood that getting camera mode is NOT part of Multiblock 2/Concept #4

**Reply by jcasman** [reply]
Requirements for Get Camera Mode When App Starts - Concept 5
- get camera mode (image or video)
- Output appears in debug console with a print statement

Second priority (optional)
Similar to iPhone shutter button
- If camera mode is image, shutter button is white
- If camera mode is video, shutter button is red

Third priority (optional)
- After image is taken, show image on same screen, below the button

Fourth priority (optional and maybe impossible)
- After video is taken, show video thumbnail on same screen, below the button. No one at Oppkey has done this. May not be possible.

**Reply by caitlynbakery** [reply]
I'll get started on it today.
